include/aws/s3/private/s3_checksums.h

```diff
@@ -43,8 +48,10 @@ struct aws_input_stream *aws_checksum_stream_new(
     struct aws_byte_buf *checksum_output);
 
 /**
- * A stream that takes in a stream, turns it into a chunk, and follows it with a aws-chunked trailer. Scanning this
- * stream will immediately fail, as it would prevent an accurate calculation of the checksum.
+ * A stream that takes in a stream, encodes it to aws_chunked. Computes a running checksum as it is read and add the
```

Comment:
Yeah, we pretend the whole stream is 1 single chunk, but it's a lawful `aws-chunked` encoding.

Reply:
isnt it kinda confusing when we call it a chunked stream, but it will ever only chunk it into 1 chunk and is primarily used to compute checksum. not a high priority now, but something to cleanup for future